src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/proxy/ProxyControl.java

```diff
@@ -355,6 +406,48 @@ public void setHTTPSampleNamingMode(int httpNamingMode) {
         setProperty(new IntegerProperty(HTTP_SAMPLER_NAMING_MODE, httpNamingMode));
     }
 
+    public void setHttpSampleNumberingMode(int httpSampleNumberingMode) {
+        
+        if (httpSampleNumberingMode == HttpSamplerNumberingMode.PREFIX.getIntValue()) {
+        	sNumberingMode = HttpSamplerNumberingMode.PREFIX.getStringMode();
+        	JMeterUtils.setProperty("proxy.number.mode", "prefix");
+        	JMeterUtils.setProperty("proxy.number.requests", "true");
+        } else if (httpSampleNumberingMode ==  HttpSamplerNumberingMode.SUFFIX.getIntValue()) {
+            // choose suffix numbering
+        	sNumberingMode =  HttpSamplerNumberingMode.SUFFIX.getStringMode();
+        	JMeterUtils.setProperty("proxy.number.mode", "suffix");
+        	JMeterUtils.setProperty("proxy.number.requests", "true");        	
+        } else if (httpSampleNumberingMode == HttpSamplerNumberingMode.NO_NUMBER.getIntValue()) {
+            // choose no numbering
+        	sNumberingMode = HttpSamplerNumberingMode.NO_NUMBER.getStringMode();
+        	JMeterUtils.setProperty("proxy.number.requests", "false");
+        }
+    }
+
+    public void setHttpSamplerNumberingStartValue(int httpSamplerNumberingStartValue) {
+        iNumberingStartValue=  httpSamplerNumberingStartValue;
+    }
+
+    public void setHttpSamplerNumberingStartForSamplerCreator(int httpSamplerNumberingStartValue) {
+        iNumberingStartValue =  httpSamplerNumberingStartValue;
+        // change the numbering start value directly to the AtomicInteger REQUEST_NUMBER in the AbstractSamplerCreator
+        SamplerCreator samplerCreator = samplerCreatorFactory.getDefaultSamplerCreator();
+        samplerCreator.setRequestNumber(httpSamplerNumberingStartValue);
+    }
+
+    public int getRequestNumberFromSamplerCreator() {
+        int iRequestNumber = 1;
+        SamplerCreator samplerCreator = samplerCreatorFactory.getDefaultSamplerCreator();
```

Comment:
I think that this assumption may not always be correct, as the `DefaultSamplerCreator` is probably the only factory, that would really work here.
On the other hand that seems to be the same with the formatting scheme changes. Hm...